src/include/sof/audio/component.h

```diff
@@ -791,7 +791,7 @@ void comp_get_copy_limits_with_lock(struct comp_buffer *source,
  */
 static inline void comp_invalidate(struct comp_dev *dev)
 {
-	if (!dev->is_shared)
+	if (dev->is_shared)
```

Comment:
@keyonjie is this correct? if a device is shared, dev is an uncache pointer and why would we pass an uncache ptr to invalidate?

Reply:
@ranj063 If the device isn't shared, there's no need to do anything with caching. And not all platforms have cache/uncache memory.

Reply:
@paulstelian97 when a device is shared, the dev pointer is modified to use the uncache alias. I dont see the need to invalidate this at all

Reply:
@ranj063 Again, this is specific to Intel. The only reason this thing works as-is is because Intel is the only one with multi-core DSP. On non-Intel platforms, including my own, there is no uncache alias to speak of. So this won't work correctly if one of the future i.MX generations has dual core/multi core DSP.

Reply:
I think @ranj063 is right: the semantics of "shared" in SOF currently is, that the object is safe to share between cores with no further cache manipulation. Other objects might or might not be shared between cores. If they are, their cache coherence has to be handled on an ad-hoc basis. I think this assumption, that "shared" objects do not need to be manually cache-synchronised is now used throughout the code. And yes, this will break on incoherent multicore DSP architectures without uncached aliases. To fix this we'd need to distinguish between "shared" (anything, potentially used by different cores) and cache-synchronous / uncached.

Reply:
```is_shared``` means that the object is shared and has to be inv/wb as per the ops. So this fix is right.

Reply:
@lgirdwood if you only got by the meaning of is_shared , yes this makes sense. But have a look at https://github.com/thesofproject/sof/blob/b1d88a88455ce2a623ecc4bada772cb257611b0d/src/audio/component.c#L443.

When a comp is made shared, its dev pointer is changed to point to the uncache alias. Does it still make sense to invalidate.

Reply:
yeah, please split this PR in two, this patch is anything but a "minor fix" if 5 of the top SOF contributors can't even agree on the meaning of "shared". Thanks!

Reply:
> this patch is anything but a "minor fix" ...

Indeed anything that inverts some cache invalidation logic seems as far from "minor" as it can be. In addition to being split into a  more visible PR, the very least this change also deserves is a step by step description of the use case(s) it fixes and of the corresponding test gaps.

> ... if 5 of the top SOF contributors can't even agree on the meaning of "shared".

I saw some tentative definitions above. They should be rescued from this "minor fixes" PR and added to the appropriate .h file. This PR for this .h file will be the perfect place to discuss and agree on the definition.

PS: I stumbled on this review by chance when looking whether some unrelated CI issue was fixed. It has, FW logs are not missing anymore from test  results.

Reply:
Hi @lgirdwood I think @ranj063 and @lyakh are correct here, thanks for raising it. The ```is_shared``` naming is as confusion as the naming of the ```.shared_data``` section. The former one means ALREADY shared/uncached, while the latter one means the data may be shared among COREs but it is in cached address (for multi-core related).

This brings the topic I talked to @ranj063 that our ```.shared_data``` section is not sharing friendly at all, users should call platform_shared_get() before R/W access to it. We have some parts (e.g. component, dmic_global) follow this rule, but some others don't, that's potential data incoherency to me.

Anyway, let me remove the wrong 2nd patch for the PR here. Thank you.

Reply:
@keyonjie @ranj063 ```is_shared``` is being used in buffer to to determine "do I need to WB or INV cache" AND is also overloaded here as "is the ptr alias in uncache region", but I agree the coherent API deals with this so we can fix as part of that update.